R/human_infection.R

```diff
@@ -295,50 +295,81 @@ calculate_treated <- function(
     )
   ])
   
+  #+++ DRUG EFFICACY +++#
+  #+++++++++++++++++++++#
+  effectively_treated_index <- bernoulli_multi_p(parameters$drug_efficacy[drugs])
+  effectively_treated <- bitset_at(seek_treatment, effectively_treated_index)
+  drugs <- drugs[effectively_treated_index]
+  n_drug_efficacy_failures <- n_treat - effectively_treated$size()
+  renderer$render('n_drug_efficacy_failures', n_drug_efficacy_failures, timestep)
+  
+  #+++ ANTIMALARIAL RESISTANCE +++#
+  #+++++++++++++++++++++++++++++++#
   if(parameters$antimalarial_resistance) {
     resistance_parameters <- get_antimalarial_resistance_parameters(
       parameters = parameters,
       drugs = drugs, 
       timestep = timestep
     )
-    unsuccessful_treatment_probability <- resistance_parameters$artemisinin_resistance_proportion * resistance_parameters$early_treatment_failure_probability
-    susceptible_to_treatment_index <- bernoulli_multi_p(p = 1 - unsuccessful_treatment_probability)
-    susceptible_to_treatment <- bitset_at(seek_treatment, susceptible_to_treatment_index)
-    drugs <- drugs[susceptible_to_treatment_index]
+    
+    #+++ EARLY TREATMENT FAILURE +++#
+    #+++++++++++++++++++++++++++++++#
+    early_treatment_failure_probability <- resistance_parameters$artemisinin_resistance_proportion * resistance_parameters$early_treatment_failure_probability
+    successfully_treated_indices <- bernoulli_multi_p(p = 1 - early_treatment_failure_probability)
+    successfully_treated <- bitset_at(effectively_treated, successfully_treated_indices)
+    n_early_treatment_failure <- effectively_treated$size() - successfully_treated$size()
+    renderer$render('n_early_treatment_failure', n_early_treatment_failure, timestep)
+    drugs <- drugs[successfully_treated_indices]
+    resistance_parameters$dt_slow_parasite_clearance <- resistance_parameters$dt_slow_parasite_clearance[successfully_treated_indices]
```

Comment:
Are we sure we want to reassign this back to `resistance_parameters`?

```suggestion
    dt_slow_parasite_clearance <- resistance_parameters$dt_slow_parasite_clearance[successfully_treated_indices]
```

Reply:
I've implemented this change as recommended here, but further down write over the new `dt_slow_parasite_clearance` as otherwise I don't indices from the bernoulli_multi_p() calls will correspond to the values they should.